js_modules/dagit/packages/core/src/workspace/ReloadAllButton.tsx

```diff
@@ -20,8 +31,37 @@ export const ReloadAllButton: React.FC<{label?: string}> = ({label = 'Reload all
   }
 
   return (
-    <Button onClick={onClick} icon={<Icon name="refresh" />} loading={reloading} intent="none">
-      {label}
-    </Button>
+    <>
+      <Button onClick={tryReload} icon={<Icon name="refresh" />} loading={reloading} intent="none">
+        {label}
+      </Button>
+      {error &&
+        (errorLocationId ? (
+          <RepositoryLocationErrorDialog
+            error={error}
+            location={errorLocationId}
+            reloading={reloading}
+            onTryReload={tryReload}
+            onDismiss={() => setIsOpen(false)}
+            isOpen={isOpen}
+          />
+        ) : (
+          <Dialog
+            icon="error"
+            title="Repository location error"
+            canEscapeKeyClose={false}
+            canOutsideClickClose={false}
+            style={{width: '90%'}}
+            isOpen={isOpen}
+          >
+            <DialogBody>
+              <PythonErrorInfo error={error} />
+            </DialogBody>
+            <DialogFooter>
+              <Button onClick={() => setIsOpen(false)}>Dismiss</Button>
+            </DialogFooter>
+          </Dialog>
+        ))}
+    </>
```

Comment:
Does mounting/unmounting based on `error` instead of using `isOpen` mean that these dialogs flash in instead of animating in?

Also, is the goal to avoid allowing `Reload again` in the second dialog?

Reply:
Ahh yeah, I think the goal here is to support showing an error that is not tied to a particular location. Let me see if we can clean this up a bit 👍  I think that switching on the error while the dialog is closed is preferable but I'll verify they animate in!